### PR TITLE
Allow custom config file for ament_uncrustify

### PR DESCRIPTION
**Pull request by Yadunund**
A minor enhancement to ament_uncrustify() to allow parsing of custom configuration file

### CODE REVIEW

File: ament_cmake_uncrustify/cmake/ament_uncrustify.cmake [hunk 1]
```diff
@@ -11,12 +11,19 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
 #
 # Add a test to check the code for compliance with uncrustify.
 #
+# The default configuration file used for uncrustify is located at
+# configuration/ament_code_style.cfg within the uncrustify directory
```
Comment (dirk-thomas) [inline ament_cmake_uncrustify/cmake/ament_uncrustify.cmake]: The file is actually within the `ament_uncrustify` directory.

Nitpick: add a trailing period.

File: ament_cmake_uncrustify/cmake/ament_uncrustify.cmake [hunk 2]
```diff
@@ -11,12 +11,19 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
```
Comment (dirk-thomas) [inline ament_cmake_uncrustify/cmake/ament_uncrustify.cmake]: Please revert this unrelated whitespace change. The empty line intentionally separates the copyright header from the function docblock.

File: ament_cmake_uncrustify/cmake/ament_uncrustify.cmake [hunk 3]
```diff
@@ -11,12 +11,19 @@
...
 #
+# The default configuration file used for uncrustify is located at
+# configuration/ament_code_style.cfg within the uncrustify directory
+# The default configuration file can be overridden by the
+# argument 'CONFIG_FILE'.
+#
 # :param TESTNAME: the name of the test, default: "uncrustify"
 # :type TESTNAME: string
+# :param CONFIG_FILE: the path of the configuration file for
+#                     uncrustify to consider
```
Comment (dirk-thomas) [inline ament_cmake_uncrustify/cmake/ament_uncrustify.cmake]: Please avoid the alignment and intend this line the same as for the `MAX_LINE_LENGTH` argument.

File: ament_cmake_uncrustify/cmake/ament_uncrustify.cmake (hunk 1 above)
Comment (Yadunund) [reply ament_cmake_uncrustify/cmake/ament_uncrustify.cmake]: Corrected.

File: ament_cmake_uncrustify/cmake/ament_uncrustify.cmake (hunk 2 above)
Comment (Yadunund) [reply ament_cmake_uncrustify/cmake/ament_uncrustify.cmake]: The whitespace change has been reverted.

File: ament_cmake_uncrustify/cmake/ament_uncrustify.cmake (hunk 3 above)
Comment (Yadunund) [reply ament_cmake_uncrustify/cmake/ament_uncrustify.cmake]: Noted. The indentation has been corrected.

**Comment by dirk-thomas** [comment]
Thanks for the enhancement.
